appcli/orchestrators.py

```diff
@@ -598,6 +599,239 @@ def __exec_command(self, command: Iterable[str]) -> CompletedProcess:
+        logger.info("HelmOrchestrator has no services to run tasks for.")
+        return None
+
+    def exec(
+        self,
+        cli_context: CliContext,
+        service_name: str,
+        command: Iterable[str],
+        stdin_input: str = None,
+        capture_output: bool = False,
+    ) -> CompletedProcess:
+        logger.info("HelmOrchestrator does not support executing arbitrary commands.")
+        return None
+
+    def get_logs_command(self) -> click.Command:
+        @click.command()
+        def log():
+            logger.info("HelmOrchestrator does not support getting logs.")
+            return None
+
+        return log
+
+    def get_additional_commands(self):
+        # TODO: AF-248: Add `kubectl`, `helm` and possibly `k9s` as commands that can be called.
+        return []
+
+    def get_name(self) -> str:
+        return "helm"
+
+    def verify_service_names(
+        self, cli_context: CliContext, service_names: tuple[str, ...]
+    ) -> bool:
+        if service_names is None or len(service_names) == 0:
+            return True
+        logger.info("HelmOrchestrator has no services.")
+        return False
+
+    def get_disabled_commands(self) -> list[str]:
+        # The `task` command is disabled because helm is only used for install/upgrade/uninstall/downgrade operations.
+        # The `init` command is disabled because it's used to initialise additional services, and this orchestrator
+        # has no services. NOTE: The `init` command will be removed in the future.
+        return ["init", "task"]
+
+    def __run_command(self, command: list[str]) -> CompletedProcess:
+        """Run the given command and return the CompletedProcess.
+
+        Args:
+            command (list[str]): The command to execute.
+
+        Returns:
+            CompletedProcess: The execution result.
+        """
+        logger.debug(f"Executing {str(command)}")
+        result = subprocess.run(command, capture_output=True)
+        if result.returncode != 0:
+            logger.error(result.stderr.decode("utf-8"))
+            raise subprocess.CalledProcessError(
+                result.returncode,
+                command,
+                result.stdout,
+                result.stderr,
+            )
+        return result
+
+    def __generate_values_args(
+        self, helm_values_dir: Path, helm_values_files_dir: Path
+    ) -> list[str | Path]:
+        """Recursively takes all the values files in the directories and generates an args array to
+        pass to helm through either `--values` or `--set-file` (depending on the location).
+        e.g:
+
+            ["--values", "values.yaml", "--set-file", "path.to.foo=path/to/foo.yaml"...
+
+        Args:
+            helm_values_dir (Path): Directory containing files to set with `--values`.
+            helm_values_files_dir (Path): Directory containing files to set with `--set-file`.
+
+        Returns:
+            list[str | Path]: The arg list.
```

Comment:
Better to just be `list[str]`. We don't really want to return `Path` in here do we? Its weird having a mix of types.
We're using this to build up the command string yeah? So we should probably be turning the Path objects into strings.

Reply:
Looks like you've changed it to be `str(Path)` but not updated method signature. Will fix.